### ISSUE TITLE
resource `meraki_networks_appliance_traffic_shaping_uplink_selection` error - on updates when state file has entry

### ISSUE THREAD

**Issue by finkjordanj** [cisco-open/terraform-provider-meraki#92]
**Prerequisites**
* [X] Have you tested the operation in the API directly?
* [X] Do you have the latest Terraform provider version?
* [X] Review the [compatibility matrix](https://github.com/meraki/terraform-provider-meraki/tree/main#compatibility-matrix) before opening an issue. 

**Describe the bug**
On an initial apply for this config the apply will complete successfully. Any updates or an state import triggers the update function to error out due to non passed in values.

Config 
```hcl
resource "meraki_networks_appliance_traffic_shaping_uplink_selection" "this" {
  network_id                     = local.this_network.id
  default_uplink                 = "wan1"
  active_active_auto_vpn_enabled = true
  load_balancing_enabled         = true
  failover_and_failback = {
    immediate = {
      enabled = true
    }
  }
}
```

Successful on initial apply

```
2024-07-02T11:47:15.107-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: BODY         :
2024-07-02T11:47:15.107-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: {
2024-07-02T11:47:15.108-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "activeActiveAutoVpnEnabled": true,
2024-07-02T11:47:15.108-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "defaultUplink": "wan1",
2024-07-02T11:47:15.109-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "loadBalancingEnabled": true,
2024-07-02T11:47:15.109-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "wanTrafficUplinkPreferences": [],
2024-07-02T11:47:15.109-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "vpnTrafficUplinkPreferences": [
2024-07-02T11:47:15.109-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       {
2024-07-02T11:47:15.110-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "trafficFilters": [
2024-07-02T11:47:15.110-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:             {
2024-07-02T11:47:15.110-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                "type": "applicationCategory",
2024-07-02T11:47:15.111-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                "value": {
2024-07-02T11:47:15.111-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                   "id": "meraki:layer7/category/16",
2024-07-02T11:47:15.111-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                   "name": "VoIP & video conferencing"
2024-07-02T11:47:15.111-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                }
2024-07-02T11:47:15.112-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:             }
2024-07-02T11:47:15.112-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          ],
2024-07-02T11:47:15.112-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "preferredUplink": "bestForVoIP"
2024-07-02T11:47:15.112-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       }
2024-07-02T11:47:15.112-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    ],
2024-07-02T11:47:15.112-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "failoverAndFailback": {
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       "immediate": {
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "enabled": true
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       }
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    }
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: }
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: ==============================================================================
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 State: NetworkID value is "L_783626335162467749", ActiveActiveAutoVpnEnabled value is true, DefaultUplink value is "wan1", FailoverAndFailback value is &{0xc00058d0a0}, LoadBalancingEnabled value is true, VpnTrafficUplinkPreferences value is <nil>, WanTrafficUplinkPreferences value is <nil>
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Resp: NetworkID value is <null>, ActiveActiveAutoVpnEnabled value is true, DefaultUplink value is "wan1", FailoverAndFailback value is &{0xc00012709e}, LoadBalancingEnabled value is true, VpnTrafficUplinkPreferences value is &[{"" 0xc000158550 "bestForVoIP" 0xc000570858}], WanTrafficUplinkPreferences value is &[]
2024-07-02T11:47:15.113-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Entre 2: fieldName Immediate
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Entre 3 {true} to field Immediate
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.114-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Entre 2: fieldName Enabled
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 2. Assigned true to field Enabled
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Sali:
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Sali:
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference NO
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 IF DeepEqual:
2024-07-02T11:47:15.115-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.116-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 Dereference
2024-07-02T11:47:15.116-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 IF DeepEqual:
2024-07-02T11:47:15.116-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:47:15 [DEBUG] result: NetworkID value is "L_783626335162467749", ActiveActiveAutoVpnEnabled value is true, DefaultUplink value is "wan1", FailoverAndFailback value is &{0xc00012720e}, LoadBalancingEnabled value is true, VpnTrafficUplinkPreferences value is &[{"" 0xc000158550 "bestForVoIP" 0xc000570858}], WanTrafficUplinkPreferences value is &[]
2024-07-02T11:47:15.116-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: tf_attribute_path=active_active_auto_vpn_enabled tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 @module=sdk.framework tf_rpc=ApplyResourceChange timestamp=2024-07-02T11:47:15.097-0500
2024-07-02T11:47:15.116-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f tf_rpc=ApplyResourceChange @module=sdk.framework tf_attribute_path=default_uplink tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection timestamp=2024-07-02T11:47:15.097-0500
2024-07-02T11:47:15.117-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: @module=sdk.framework tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki tf_attribute_path=failover_and_failback.immediate.enabled tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 timestamp=2024-07-02T11:47:15.097-0500
2024-07-02T11:47:15.117-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: @module=sdk.framework tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection tf_attribute_path=failover_and_failback.immediate tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 timestamp=2024-07-02T11:47:15.097-0500
2024-07-02T11:47:15.118-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection @module=sdk.framework tf_attribute_path=failover_and_failback tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 timestamp=2024-07-02T11:47:15.097-0500
2024-07-02T11:47:15.118-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection tf_rpc=ApplyResourceChange @module=sdk.framework tf_attribute_path=load_balancing_enabled tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki timestamp=2024-07-02T11:47:15.098-0500
2024-07-02T11:47:15.118-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: Value switched to prior value due to semantic equality logic: tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki tf_req_id=d82a8a9f-4e69-1112-b935-4bb739c1779f @caller=github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwschemadata/value_semantic_equality.go:87 @module=sdk.framework tf_attribute_path=network_id tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection tf_rpc=ApplyResourceChange timestamp=2024-07-02T11:47:15.098-0500
meraki_networks_appliance_traffic_shaping_uplink_selection.this: Creation complete after 1s
2024-07-02T11:47:15.158-0500 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-07-02T11:47:15.160-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-07-02T11:47:15.162-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/cisco-open/meraki/0.2.5-alpha/linux_amd64/terraform-provider-meraki_v0.2.5-alpha id=6253
2024-07-02T11:47:15.162-0500 [DEBUG] provider: plugin exited

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

Error on second apply only change is failover/Failback
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # meraki_networks_appliance_traffic_shaping_uplink_selection.this will be updated in-place
  ~ resource "meraki_networks_appliance_traffic_shaping_uplink_selection" "this" {
      ~ failover_and_failback          = {
          ~ immediate = {
              ~ enabled = true -> false
            }
        }
        # (6 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
2024-07-02T11:52:13.960-0500 [DEBUG] command: asking for input: "\nDo you want to perform these actions?"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

meraki_networks_appliance_traffic_shaping_uplink_selection.this: Modifying...
2024-07-02T11:52:25.744-0500 [INFO]  Starting apply for meraki_networks_appliance_traffic_shaping_uplink_selection.this
2024-07-02T11:52:25.744-0500 [DEBUG] skipping FixUpBlockAttrs
2024-07-02T11:52:25.745-0500 [DEBUG] meraki_networks_appliance_traffic_shaping_uplink_selection.this: applying the planned Update change
2024-07-02T11:52:26.207-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: 2024/07/02 11:52:26
2024-07-02T11:52:26.207-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: ==============================================================================
2024-07-02T11:52:26.207-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: ~~~ REQUEST ~~~
2024-07-02T11:52:26.208-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: PUT  /api/v1/networks/L_783626335162467749/appliance/trafficShaping/uplinkSelection  HTTP/1.1
2024-07-02T11:52:26.208-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: HOST   : api.meraki.com
2024-07-02T11:52:26.208-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: HEADERS:
2024-07-02T11:52:26.208-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Accept: application/json
2024-07-02T11:52:26.208-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Authorization: Bearer ****30e10
2024-07-02T11:52:26.209-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Content-Type: application/json
2024-07-02T11:52:26.209-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   User-Agent: go-meraki/1.44.1 MerakiTerraform/1.47.0 Cisco
2024-07-02T11:52:26.209-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: BODY   :
2024-07-02T11:52:26.209-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: {
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "activeActiveAutoVpnEnabled": true,
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "defaultUplink": "wan1",
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "failoverAndFailback": {
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       "immediate": {
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "enabled": false
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       }
2024-07-02T11:52:26.210-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    },
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "loadBalancingEnabled": true,
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "vpnTrafficUplinkPreferences": [
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       {
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "performanceClass": {},
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "preferredUplink": "bestForVoIP",
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          "trafficFilters": [
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:             {
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                "type": "applicationCategory",
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                "value": {
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                   "destination": {},
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                   "id": "meraki:layer7/category/16",
2024-07-02T11:52:26.211-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                   "source": {}
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:                }
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:             }
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:          ]
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       }
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    ]
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: }
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: ------------------------------------------------------------------------------
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: ~~~ RESPONSE ~~~
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: STATUS       : 400 Bad Request
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: PROTO        : HTTP/2.0
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: RECEIVED AT  : 2024-07-02T11:52:26.2071927-05:00
2024-07-02T11:52:26.212-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: TIME DURATION: 458.0872ms
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: HEADERS      :
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Cache-Control: no-cache, no-store, max-age=0, must-revalidate
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Content-Type: application/json; charset=utf-8
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Date: Tue, 02 Jul 2024 16:52:26 GMT
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Expires: Fri, 01 Jan 1990 00:00:00 GMT
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Pragma: no-cache
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   Strict-Transport-Security: max-age=31536000; includeSubDomains
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   X-Content-Type-Options: nosniff
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   X-Frame-Options: sameorigin
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   X-Request-Id: 2eea30c0e0b5712db351c86bb39b3987
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   X-Robots-Tag: none
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   X-Runtime: 0.075250
2024-07-02T11:52:26.213-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:   X-Xss-Protection: 1; mode=block
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: BODY         :
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: {
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    "errors": [
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:       "'performanceClass' is missing the following required properties: 'type'"
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha:    ]
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: }
2024-07-02T11:52:26.214-0500 [DEBUG] provider.terraform-provider-meraki_v0.2.5-alpha: ==============================================================================
2024-07-02T11:52:26.214-0500 [ERROR] provider.terraform-provider-meraki_v0.2.5-alpha: Response contains error diagnostic: diagnostic_detail="error with operation UpdateNetworkApplianceTrafficShapingUplinkSelection" diagnostic_severity=ERROR tf_proto_version=6.6 tf_req_id=d4fdb40c-048f-3081-e2ac-625ab6f8569e tf_resource_type=meraki_networks_appliance_traffic_shaping_uplink_selection tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_summary="Failure when executing UpdateNetworkApplianceTrafficShapingUplinkSelection" tf_provider_addr=registry.terraform.io/CiscoDevNet/meraki timestamp=2024-07-02T11:52:26.209-0500
2024-07-02T11:52:26.217-0500 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-07-02T11:52:26.217-0500 [ERROR] vertex "meraki_networks_appliance_traffic_shaping_uplink_selection.this" error: Failure when executing UpdateNetworkApplianceTrafficShapingUplinkSelection
╷
│ Error: Failure when executing UpdateNetworkApplianceTrafficShapingUplinkSelection
│
│   with meraki_networks_appliance_traffic_shaping_uplink_selection.this,
│   on router.tf line 73, in resource "meraki_networks_appliance_traffic_shaping_uplink_selection" "this":
│   73: resource "meraki_networks_appliance_traffic_shaping_uplink_selection" "this" {
│
│ error with operation UpdateNetworkApplianceTrafficShapingUplinkSelection
╵
2024-07-02T11:52:26.225-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-07-02T11:52:26.226-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/cisco-open/meraki/0.2.5-alpha/linux_amd64/terraform-provider-meraki_v0.2.5-alpha id=6433
2024-07-02T11:52:26.226-0500 [DEBUG] provider: plugin exited
```

**Expected behavior**
Change or import of configuration into terraform apply successfully and no require a performance class be passed in.

**Screenshots**
N/A

**Environment (please complete the following information):**
* Meraki Dashboard version: 
* Terraform version: 1.9.0
* Meraki provider version: 0.2.5-alpha
* OS Version: ubuntu22.04

**Additional context**
Currently found when also trying to do a import of the configuration from the dashboard into terraform gets same error guessing due to what is being passed in vs imported also thinking this update call is needed.

**Comment by fmunozmiranda** [comment]
@finkjordanj please try it again with new version and update us.

**Comment by finkjordanj** [comment]
Resolved with 0.2.6-alpha
